packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -146,6 +138,16 @@ class CollectionData with Names {
       },
     ).toList();
 
+    final hasJsonSerializable =
+        jsonSerializableChecker.hasAnnotationOf(collectionTargetElement);
+    // Freezed classes are also JsonSerializable
+    if (!hasJsonSerializable && !hasFreezed) {
```

Comment:
Using Freezed or json-serializable wasn't required before. This is a breaking change

Reply:
No we shouldn't require it.

Reply:
Did you say here you were fine requiring it or am I misreading?

https://github.com/firebase/flutterfire/pull/11365#issuecomment-1681427822

Reply:
We require the flag if json_serializable is used. But if it isn't used, we don't.

Reply:
What are the merits for using custom json serialization? Is there anything that json_serializable can't handle with converters? The only other option I see as viable is have people implement their own `perFieldToJson` methods if they don't want to use json_serializable. Otherwise we would have to maintain two different code-gen options. One that uses json_serializable with perFieldToJson that supports all types and one without that doesn't support all types. The second option does not sound great to me. What are your thoughts?

Reply:
I think that's fair. We can add support for it back if necessary later.